x-pack/plugin/identity-provider/src/main/java/org/elasticsearch/xpack/idp/saml/sp/SamlServiceProviderDocument.java

```diff
@@ -52,14 +54,15 @@
 
     public static class Privileges {
         public String resource;
-        public Set<String> rolePatterns = Set.of();
+        // we use a sorted set so that the order is consistent in XContent APIs
+        public SortedSet<String> rolePatterns = new TreeSet<>();
```

Comment:
I had the same thought stream. I don't think there's really any issue here. Yes, we're sorting the set where it might not be needed, but 99% of the time it will have 1 item in it, so the impact is effectively zero in real world usage.